dmt/main/tests/factories.py

```diff
@@ -15,6 +18,23 @@ class Meta:
     email = factory.Sequence(lambda n: 'user{0}@columbia.edu'.format(n))
     grp = False
     password = ""
+    user = factory.SubFactory('dmt.main.tests.factories.UserFactory',
+                              userprofile=None)
+
+
+class UserFactory(factory.DjangoModelFactory):
+    class Meta:
+        model = User
+
+    username = factory.Sequence(lambda n: "user_%d" % n)
+    userprofile = factory.RelatedFactory(UserProfileFactory, 'user')
+
+    @classmethod
+    def _generate(cls, create, attrs):
+        post_save.disconnect(create_user_profile, User)
+        user = super(UserFactory, cls)._generate(create, attrs)
+        post_save.connect(create_user_profile, User)
+        return user
```

Comment:
http://factoryboy.readthedocs.org/en/latest/recipes.html#example-django-s-profile

Basically, since there's a handler attached to the `post_save` on `User` objects to automatically instantiate `UserProfile` objects, factory boy needs to disconnect that signal, make the `User`, then re-attach it.